### ISSUE TITLE
Warning on save

### ISSUE THREAD

**Issue by OfekShilon** [rstudio/rstudio#7001]
### System details

    RStudio Edition : Server
    RStudio Version : 1.2.5042
    OS Version      : Ubuntu 14.04.5 LTS
    R Version       : 3.4.0

### Steps to reproduce the problem

Run from the RStudio console:
> op<-options()
> save(list = "op", file = "/home/you/whatever")

### Describe the problem in detail
The output is:
> Warning message:
> In save(list = "op", file = "/home/you/whatever") :
>  'package:datasets' may not be available when loading

### Describe the behavior you expected
A clear console.
This is a toy example of a problem we see in our unit tests. These are configured to treat warnings as errors, and thereby fail - *only* when run from RStudio.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by OfekShilon** [comment]
It seems a similar issue was discussed here: https://stackoverflow.com/questions/27623901/r-warning-packagestats-may-not-be-available-when-loading

**Comment by kevinushey** [comment]
More minimal reproducible example:

```
askpass <- getOption("askpass")
save(askpass, file = tempfile())
```

This is because we directly bind our internal functions to some of these options, e.g. here:

https://github.com/rstudio/rstudio/blob/a1e2dcec8dc248158ca05b6a52d5cc0eeb8dab57/src/cpp/r/R/Options.R#L90-L91

And those functions belong to our RStudio 'tools' environment, which is on the search path and so "inherits" from other environments:

```
> pryr::parenvs(askpass)
   label                            name               
1  <environment: 0x7ff08dfbb638>    "tools:rstudio"    
2  <environment: package:stats>     "package:stats"    
3  <environment: package:graphics>  "package:graphics" 
4  <environment: package:grDevices> "package:grDevices"
5  <environment: package:utils>     "package:utils"    
6  <environment: package:datasets>  "package:datasets" 
7  <environment: package:methods>   "package:methods"  
8  <environment: 0x7ff0ad82f7c8>    "Autoloads"        
9  <environment: base>              ""                 
10 <environment: R_EmptyEnv>        "
```

We should bind these to standalone functions that call the underying `.rs.` helper functions to avoid this.

**Comment by OfekShilon** [comment]
Just noticed this is a duplicate: https://github.com/rstudio/rstudio/issues/3946
Sorry. Feel free to close one as duplicate of the other.

**Comment by dfalty** [comment]
Verified fixed using steps given on version 1.4.874.
